docs/ops/postgres/add-database.rst

```diff
@@ -0,0 +1,136 @@
+
+==========================
+Add the database to charts
+==========================
+
+First, create the entries in ``charts``.  Go to the
+``charts/postgres/templates`` directory, and edit ``deployment.yaml`` to
+add the new database/password entry.  You should copy an existing
+entry, and it should look like this:
+
+   .. code-block:: yaml
+
+      {{- with .Values.exposurelog_db }}
+      - name: VRO_DB_EXPOSURELOG_USER
+        value: {{ .user }}
+      - name: VRO_DB_EXPOSURELOG_DB
+        value: {{ .db }}
+      - name: VRO_DB_EXPOSURELOG_PASSWORD
+        valueFrom:
+          secretKeyRef:
+            name: postgres
+            key: exposurelog_password
+      {{- end }}
+
+Once you've done that, make sure you increment the chart version number in
+``charts/postgres/Chart.yaml``.
+
+===========================
+Add the database to phalanx
+===========================
+
+Next, tackle ``phalanx``.  First, add the password entry to Phalanx's
+installer, so the next time a new cluster is deployed or an extant
+cluster is redeployed, the password will be created.  This belongs in
+``installer/generate_secrets.py`` in the ``_postgres()`` method.
+
+Typically we use passwords that are ASCII representations of random
+32-byte hexadecimal sequences.  The passwords for all the non-root
+Postgres users already look like that, so copying an existing line
+and changing the name to reflect your service is usually correct:
+
+   .. code-block:: python
+    
+      self._set_generated("postgres", "exposurelog_password", secrets.token_hex(32))
+
+Make the Phalanx ``services/postgres/Chart.yaml`` entry depend on the
+new chart version you earlier created.
+
+Finally, go edit the postgres ``values-<env>.yaml`` files and add
+a section for your new database with appropriate ``user`` and ``db``
+entries:
+
+   .. code-block:: yaml
+
+      exposurelog_db:
+        user: 'exposurelog'
+        db: 'exposurelog'
+
+Now start the PR and review process.  However, there is a step you still
+must do before you can synchronize the updated services: put the
+password into Vault so it appears in the postgres secrets.
+
+================================
+Manually add the secret to Vault
+================================
+
+Since you have already added generation of the password to the
+installer, you could just generate new secrets for each environment and
+push them into Vault.  That, however, would require that you restart
+everything with randomly-generated passwords, and that's a fairly
+disruptive operation, so you probably are better off manually injecting
+just your new password.
+
+* Consult ``1Password`` and retrieve the appropriate vault write token for
+  the instance you're working with from ``vault_keys.json``.
+* Set up your environment: ``export VAULT_ADDR=vault.lsst.codes ; export
+  VAULT_FORMAT=json ; export VAULT_TOKEN=<retrieved-token>``
+* Run ``vault kv get secret/k8s_operator/<instance>/postgres >
```

Comment:
I think you can simplify these steps by suggesting simply `vault kv patch secret/k8s_operator/<instance>/postgres exposurelog_password=<password>`

Reply:
Oh!  Yes, that would be simpler than what I've been doing.

Reply:
Done.